app/models/user.rb

```diff
@@ -197,4 +200,16 @@ def self.from_omniauth(auth, current_user)
 
     authorization.user
   end
+
+  def self.twitter_mastodon_mapping
+    Rails.cache.fetch('/twitter_mastodon_mapping') do
+      # TODO consider storing the twitter_handle in DB
+      # user.twitter_handle sends an individual request
+      # https://developer.twitter.com/en/docs/twitter-api/users/lookup/api-reference/get-users-by-username-username
+      # rate limit: 900 requests per 15-min window shared among all users of your app
+      User.all.each do |user|
+        [user.twitter_handle, "@#{user.mastodon_id}"]
```

Comment:
I checked the Twitter API and it seams the translation from Twitter IDs to `twitter_handle` does not allow for a batch request. With many accounts registered on a crossposer instance, this call to `#twitter_handle` in a loop can potentially be heavy.

I don't expect more than 200 accounts for the next months. So it's acceptable for the time being. However, I wonder if the twitter_handle should not be put in the local database alongside the twitter ID.

Reply:
I do think it should go in the database, it would make more sense. Then you don't have to deal with the cache, it should go in the `Authorization` table, so that it gets cleared automatically if the user changes the twitter account they linked :)